packages/api/src/db/comment.ts

```diff
@@ -0,0 +1,102 @@
+import {RichTextNode} from '../graphql/richText'
+import {SortOrder, Limit, InputCursor, ConnectionResult} from './common'
+
+export enum CommentState {
+  Approved = 'Approved',
+  PendingApproval = 'PendingApproval',
+  PendingUserChanges = 'PendingUserChanges',
+  Rejected = 'Rejected'
+}
+
+export enum CommentRejectionReason {
+  Misconduct = 'Misconduct',
+  Spam = 'Spam'
+}
+
+export enum CommentAuthorType {
+  Team = 'Team',
+  Author = 'Author',
+  VerifiedUser = 'VerifiedUser'
+}
+
+export enum CommentItemType {
+  Article = 'Article',
+  Page = 'Page'
+}
+
+export interface CommentData {
+  readonly userID: string
+
+  readonly itemID: string
+  readonly itemType: CommentItemType
+
+  readonly parentID?: string
+}
+
+export interface Comment extends CommentData {
+  readonly id: string
+
+  readonly authorType: CommentAuthorType
+
+  readonly revisions: CommentRevision[]
+
+  readonly state: CommentState
+
+  readonly rejectionReason?: CommentRejectionReason
+
+  readonly createdAt: Date
+  readonly modifiedAt: Date
+}
+
+export interface CommentRevision {
+  text: RichTextNode[]
+  createdAt: Date
+}
+
+export interface PublicComment extends CommentData {
+  readonly id: string
+
+  readonly authorType: CommentAuthorType
```

Comment:
You're using `authorType` in `PublicComment` and `Comment` so you can move it to `CommentData`

Reply:
fixed